legacy_packages/react/src/evm/hooks/wallets/useEmbeddedWallet.ts

```diff
@@ -200,8 +200,58 @@ export function useEmbeddedWalletUserEmail(): UseQueryResult<
     [wallet?.walletId, "embeddedWallet-email"],
     () => {
       if (wallet && wallet.walletId === walletIds.embeddedWallet) {
-        return (wallet as EmbeddedWallet).getEmail();
+        return (wallet as EmbeddedWallet).getEmail() ?? "";
+      }
+      return "";
+    },
+    {
+      retry: false,
+      enabled: wallet?.walletId === walletIds.embeddedWallet,
+    },
+  );
+
+  // Invalidate the query when the wallet changes
+  useEffect(() => {
+    queryClient.invalidateQueries([wallet?.walletId, "embeddedWallet-email"]);
+  }, [wallet, queryClient]);
```

Comment:
this should be handled inside the `useQuery` already since it has the `walletId` as part of the query key? why do we need this extra `useEffect`?